dbt/include/sqlserver/macros/adapters.sql

```diff
@@ -36,9 +45,24 @@
   {% endcall %}
 {% endmacro %}
 
-{% macro sqlserver__drop_schema(database_name, schema_name) -%}
+{% macro sqlserver__drop_schema(relation) -%}
+  {%- set tables_in_schema_query %}
+      SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES
+      WHERE TABLE_SCHEMA = '{{ relation.schema }}'
+  {% endset %}
+  {% set tables_to_drop = run_query(tables_in_schema_query).columns[0].values() %}
+  {% for table in tables_to_drop %}
+    {%- set schema_relation = adapter.get_relation(database=relation.database,
+                                               schema=relation.schema,
+                                               identifier=table) -%}
+    {% do drop_relation(schema_relation) %}
+  {%- endfor %}
+
   {% call statement('drop_schema') -%}
-    drop schema if exists {{ relation.without_identifier().schema }}
+      IF EXISTS (SELECT * FROM sys.schemas WHERE name = '{{ relation.schema }}')
+      BEGIN
+      EXEC('DROP SCHEMA {{ relation.schema }}')
+      END
```

Comment:
@mikaelene here is the change we made that @jtcohen6 mentioned here https://github.com/mikaelene/dbt-sqlserver/issues/54#issuecomment-700786503.

In our adapters we never use `drop_schema()`, nor does `dbt-integration-tests` so we never new it was problematic.